### PR TITLE
Add support for VM_EVENT_FLAG_SET_REGISTERS on Xen 4.7

### PR DESCRIPTION
**Pull request by tklengyel**

### CODE REVIEW

**Comment by tklengyel** [comment]
Retest this please.

File: libvmi/driver/xen/xen_events.c [hunk 1]
```diff
@@ -246,9 +249,10 @@ status_t process_interrupt_event(vmi_instance_t vmi,
         event->interrupt_event.gla = req->data.regs.x86.rip;
         event->interrupt_event.intr = intr;
         event->interrupt_event.reinject = -1;
-#if VM_EVENT_INTERFACE_VERSION >= 2
-        event->interrupt_event.insn_length = req->u.software_breakpoint.insn_length;
-#endif
+
+        if ( req->version >= 2 )
```
Comment (smaresca) [inline libvmi/driver/xen/xen_events.c]: This is fine to merge as-is, but I'd like to see a follow-up PR with comments in any place where there's a differing code behavior based upon the vm_event version or the xen maj/minor version #s. 

Previously, the preprocessor statement made the reason for the insn_length statement clear just by its name. Now, the line really can't be taken at face value until an unfamiliar developer learns elsewhere what the version member represents.

Comment (tklengyel) [inline libvmi/driver/xen/xen_events.c]: Sure, I'll add the comments in a follow-up PR.

Comment (smaresca) [inline libvmi/driver/xen/xen_events.c]: Just a note for posterity: this modification is not related to support for VM_EVENT_FLAG_SET_REGISTERS, but updated in passing/while rebasing.

**Comment by smaresca** [comment]
ok, +1 after review from me
